### ISSUE TITLE
Support generic types in JSON configuration (Silverback.Integration.Configuration)

### ISSUE THREAD

**Issue by BEagle1984** [BEagle1984/silverback#77]
As reported in the discussion of the issue #75 the Silverback.Integration.Configuration package currently doesn't handle generic types.
This is the case reported in the issue #75:
```json
  "Silverback": {
    "Using": [
      "Silverback.Integration.Kafka",
      "Silverback.Core.Model",
      "Silverback.Integration.Kafka.SchemaRegistry",
      "Super.GlobalPlatform.Authorizer.Intregation.Contracts"
    ],
    "Outbound": [
      {
        "MessageType": "RequestReceivedEvent",
        "Endpoint": {
          "Type": "KafkaProducerEndpoint",
          "Name": "cd-authorizer-received-requests",
          "Configuration": {
            "BootstrapServers": "PLAINTEXT://localhost:9092",
            "ClientId": "cd-ms-authorizer"
          },
          "Serializer": {
            "Type": "AvroMessageSerializer`1",
            "SchemaRegistryConfig": {
              "Url": "http://localhost:8081/"
            },
            "AvroSerializerConfig": {
              "AutoRegisterSchemas": true
            }
          }
        }
      }
    ]
  }
```

```
      Application startup exception
Silverback.Messaging.Configuration.SilverbackConfigurationException: Error in Outbound configuration section. See inner exception for details.
 ---> System.ArgumentException: Cannot create an instance of Silverback.Messaging.Serialization.AvroMessageSerializer`1[TMessage] because Type.ContainsGenericParameters is true.
```

**Comment by BEagle1984** [comment]
I'm currently thinking about discontinuing `Silverback.Integration.Configuration` because:
1. the configuration gets very verbose when using multiple endpoints
1. it can get complex with a lot of nesting and it is very error prone since there's no schema validation in place
1. you don't get warned about methods marked as obsolete, since it doesn't get compiled (I use the `Obsolete` attribute a lot to signal the option that have been left there only for backward compatibility but will be removed in the long run)
1. it will anyway be impossible to support all constructs (e.g. lambda expressions)

Considering all the above point I really wonder if it's worth the effort to keep supporting this.

**Comment by alefcarlos** [comment]
I think you should discontinue this feature, other problem is that you can't get build errors if you change the event or assembly names.

**Comment by msallin** [comment]
Started with this but switched to in code config, as you know. Never regretted the in code approach so far. However, until now, we have only very simple scenarios.

**Comment by BEagle1984** [comment]
Silverback.Integration.Configuration is deprecated and will not be shipped with the upcoming version 3.0.0
